contracts/token/ERC721/ERC721.sol

```diff
@@ -386,8 +392,12 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * Emits an {Approval} event.
      */
     function _approve(address to, uint256 tokenId) internal virtual {
+        address owner = ownerOf(tokenId);
```

Comment:
IMO reverting if the token doesn't exist is the right thing here. Setting an approval prior to minting seems like a bug more than a feature to me.

Reply:
> Setting an approval prior to minting seems like a bug more than a feature to me.

It depends on how you brand it 😎 (I'm joking)

Jokes aside, I agree, I was trying to make a case in which the `delete tokenApprovals[tokenId]` in `_update` can be used for skipping an approval deletion.

Let's keep it as it is right now, but we'll need to take a double look into the interface descriptions. See https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4377#discussion_r1261806819

Reply:
I'm still uncomfortable with an internal function using a public function. Here's another idea.

There's another comment asking for the `_exists(uint256)` function (https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4377/files#r1261760016), I think we can make a `_checkExists(address owner, uint256 tokenId)` that doesn't check actual ownership (as in `_isApproved`) and reverts with `ERC721NonexistentToken`. We're making that check in multiple places and make sense having a function for that.

Reply:
A `_checkExists` that does not check the value given, and just reverts (with some custom) error if the first input is 0 doesn't feel like an helper function I'd want.

There are already so many in ERC721, i'd like the number to go down, not up.

@frangio WDYT?

Reply:
Personally I would also prefer to avoid such a function. Though I think there is some credit to this concern:
>I'm still uncomfortable with an internal function using a public function.
The way that I justify it is that `ownerOf` is "expected to be" simply `_ownerOf` + nonzero check, and I can't think of any actual issues this (internal using a public function) might cause.

Reply:
Solving agreeing with keeping it.